src/rpc/blockchain.cpp

```diff
@@ -599,6 +599,16 @@ UniValue getblockheader(const UniValue &params, bool fHelp)
     return blockheaderToJSON(pblockindex);
 }
 
+// Allows passing int instead of bool
+static bool is_param_trueish(const UniValue &param)
```

Comment:
We have a function that does something similar called "IsStringTrue" in util.cpp.  It looks like it only accepts "1" for true though.  We should call the same thing for all of our RPC booleans so our interface is consistent.

Should we port this code into IsStringTrue so that number > 1 are True?
I have a weak opinion on this -- I'd prefer it to not be the case for the RPC.  But since other clients are doing it, we probably should as well for compatibility reasons... I'll leave that to your discretion.

Reply:
I think we should attempt to stay similar to other clients for the RPC interface so that software written for other clients doesn't need to be modified. So similarity over consistency. That includes not being too permissive with the input values, and IsStringTrue is very permissive, allowing On oN True enaBLE etc etc. When we do a flatbuffer RPC interface we can get more creative.

btw, looks like IsStringTrue has a bug, it's throwing a string and not an exception.
>     // If not found:
>    throw std::string("IsStringTrue() was passed an Invalid string");